Detectors/TRD/workflow/src/TRDEventDisplayFeedSpec.cxx

```diff
@@ -0,0 +1,290 @@
+    float localY = cTracklet.getY();
+    // Slope of raw tracklet (key dyDxAN in JSON)
+    float rawDyDx = tracklet.getUncalibratedDy() / mGeo->cdrHght();
+    // Slope of calibrated tracklet
+    float dyDx = cTracklet.getDy() / mGeo->cdrHght();
+
+    json trackletJson = {
+      {"id", trackletId},
+      {"stk", stack},
+      {"sec", sector},
+      {"lyr", layer},
+      {"row", tracklet.getPadRow()},
+      {"trk", trackId},
+      {"lY", localY},
+      {"dyDx", dyDx},
+      {"dyDxAN", rawDyDx}};
+
+    if (trackId == "null") {
+      trackletJson["trk"] = nullptr;
+    }
+
+    trackletArray.push_back(trackletJson);
+  }
+  return trackletArray;
+}
+
+void TRDEventDisplayFeedSpec::writeDigits(gsl::span<const Digit> digits, int iEvent)
+{
+  const auto& trigRec = mTrigRecs[iEvent];
+  for (int det = 0; det < constants::MAXCHAMBER; det += 6) {
+    if (mUsedDetectors[det]) {
+      int sector = mGeo->getSector(det);
+      int stack = mGeo->getStack(det);
+
+      json digitsJson = {
+        {"evid", iEvent},
+        {"lyrs", json::array()}};
+
+      for (int iLayer = 0; iLayer < 6; iLayer++) {
+        json layerJson = {
+          {"lyr", iLayer},
+          {"pads", json::array()}};
+
+        for (int iDigit = trigRec.getFirstDigit(); iDigit < trigRec.getFirstDigit() + trigRec.getNumberOfDigits(); ++iDigit) {
+          Digit digit = digits[iDigit];
+
+          int detector = digit.getDetector();
+
+          if (detector == det + iLayer) {
+            // Digits are in stack with track
+            int row = digit.getPadRow();
+            int col = digit.getPadCol();
+
+            json padJson = {
+              {"row", row},
+              {"col", col},
+              {"tbins", json::array()}};
+
+            // Event display draws 24 time bins
+            int nTimebins = 25;
+            int t = 0;
+            for (auto adc : digit.getADC()) {
+              padJson["tbins"].push_back(adc);
+              if (++t == nTimebins) {
+                break;
+              }
+            }
+            layerJson["pads"].push_back(padJson);
+          }
+        }
+        digitsJson["lyrs"].push_back(layerJson);
+      }
+      std::ofstream digitsOut(Form("../alice-trd-event-display/data/o2/E%d.%d.%d.json", iEvent, sector, stack));
+      digitsOut << digitsJson.dump(4);
+    }
+  }
+} // namespace trd
+
+void TRDEventDisplayFeedSpec::run(o2::framework::ProcessingContext& pc)
+{
+  LOG(info) << "Running event display feed...";
+
+  json jsonData = json::array();
+
+  auto tracks = pc.inputs().get<gsl::span<TrackTRD>>("trdtracks");
+  auto tracklets = pc.inputs().get<gsl::span<Tracklet64>>("trdtracklets");
+  auto digits = pc.inputs().get<gsl::span<Digit>>("trddigits");
+
+  mTrigRecs = pc.inputs().get<gsl::span<TriggerRecord>>("trdtriggerrec");
+  auto trackTrigRecs = pc.inputs().get<gsl::span<TrackTriggerRecord>>("tracktriggerrec");
+
+  // Display only first 5 events
+  int nEvents = 5;
```

Comment:
This should be compared to `mTrigRecs.size()` in order to avoid a segfault if nEvents is too large:
`int nEvents = std::min(mTrigRecs.size(), nEventsMax)`
One could also make `nEventsMax` configurable from command line.